internal/configure.go

```diff
@@ -198,28 +240,71 @@ func ConfigureUploader() (uploader *Uploader, err error) {
 // In case no configuration in environment variables found, return `<nil>` value.
 func ConfigureCrypter() crypto.Crypter {
 	loadPassphrase := func() (string, bool) {
-		return LookupConfigValue("WALG_PGP_KEY_PASSPHRASE")
+		return GetSetting(PgpKeyPassphraseSetting)
 	}
 
-	// key can be either private (for download) or public (for upload)	
-	armoredKey, isKeyExist := LookupConfigValue("WALG_PGP_KEY")
-
-	if isKeyExist {
+	// key can be either private (for download) or public (for upload)
+	if armoredKey, isKeyExist := GetSetting(PgpKeySetting); isKeyExist {
 		return openpgp.CrypterFromKey(armoredKey, loadPassphrase)
 	}
 
 	// key can be either private (for download) or public (for upload)
-	armoredKeyPath, isPathExist := LookupConfigValue("WALG_PGP_KEY_PATH")
-
-	if isPathExist {
+	if armoredKeyPath, isPathExist := GetSetting(PgpKeyPathSetting); isPathExist {
 		return openpgp.CrypterFromKeyPath(armoredKeyPath, loadPassphrase)
 	}
 
-	keyRingID := GetSettingValue("WALE_GPG_KEY_ID")
-
-	if keyRingID != "" {
+	if keyRingID, ok := GetWaleCompatibleSetting(GpgKeyIDSetting); ok {
+		tracelog.WarningLogger.Printf(`
+You are using deprecated functionality that uses an external gpg library.
+It will be removed in next major version.
+Please set GPG key using environment variables %s or %s.\n`, PgpKeySetting, PgpKeyPathSetting)
 		return openpgp.CrypterFromKeyRingID(keyRingID, loadPassphrase)
 	}
 
 	return nil
-}
\ No newline at end of file
+}
+
+func GetMaxDownloadConcurrency() (int, error) {
+	return GetMaxConcurrency(DownloadConcurrencySetting)
+}
+
+func GetMaxUploadConcurrency() (int, error) {
+	return GetMaxConcurrency(UploadConcurrencySetting)
+}
+
+// This setting is intentionally undocumented in README. Effectively, this configures how many prepared tar Files there
+// may be in uploading state during backup-push.
+func GetMaxUploadQueue() (int, error) {
+	return GetMaxConcurrency(UploadQueueSetting)
+}
+
+func GetMaxUploadDiskConcurrency() (int, error) {
+	return GetMaxConcurrency(UploadDiskConcurrencySetting)
+}
+
+func GetMaxConcurrency(key string) (int, error) {
+	conc := GetSettingWithDefault(key)
+	con, err := strconv.Atoi(conc)
+
+	if err != nil {
+		return 1, err
+	}
+	if con < 1 {
+		return 1, NewInvalidConcurrencyValueError(con)
+	}
+	return con, nil
+}
+
+func GetSentinelUserData() interface{} {
```

Comment:
I guess the format of sentinel is defined. So there could be a better way than returning interface{}.